### PR TITLE
chore: remove cosmoscmd

### PR DESCRIPTION
**Pull request by PaddyMc**
### Description

- Removed all instances of `cosmoscmd` from the node
- Please comment on the use of setting the config in the root.go `initRootCmd` function => https://github.com/elesto-dao/elesto/pull/216/files#diff-49629165448cdce392d25d6a9ba077eabeef0ddc15446cba298dabd9c921dae6R132 . To me this seems not to follow the same pattern as other chains.

### How to test

- `make test`
- `make start-dev`
- `make seed`
- `make test-sim-benchmark`

#### On upgrading to 0.46.0

I think we need to wait until IBC-GO is updated to the latest SDK version to upgrade our node. This issue arises when upgrading:
```
github.com/elesto-dao/elesto/v2/app imports
	github.com/cosmos/ibc-go/v3/modules/apps/27-interchain-accounts tested by
	github.com/cosmos/ibc-go/v3/modules/apps/27-interchain-accounts.test imports
	github.com/cosmos/ibc-go/v3/testing/simapp imports
	github.com/cosmos/ibc-go/v3/testing/simapp/params imports
	github.com/cosmos/cosmos-sdk/x/auth/legacy/legacytx: module github.com/cosmos/cosmos-sdk@latest found (v0.46.0), but does not contain package github.com/cosmos/cosmos-sdk/x/auth/legacy/legacytx
```
Any thoughts on this? @noandrea @gsora 

#### On using Osmosis IAVL

After adding this to `go.mod`:
```
replace (
	// Our cosmos-sdk branch is:  https://github.com/osmosis-labs/cosmos-sdk v0.45.0x-osmo-v7
	github.com/cosmos/cosmos-sdk => github.com/osmosis-labs/cosmos-sdk v0.45.1-0.20220611234148-f59c62f19567
	// Use Osmosis fast iavl
	github.com/cosmos/iavl => github.com/osmosis-labs/iavl v0.17.3-osmo-v7
)
```

There was this issue:
```
# github.com/elesto-dao/elesto/v2/cmd/elestod/cmd
cmd/elestod/cmd/root.go:271:11: undefined: baseapp.SetSnapshotStore
cmd/elestod/cmd/root.go:272:11: undefined: baseapp.SetSnapshotInterval
cmd/elestod/cmd/root.go:273:11: undefined: baseapp.SetSnapshotKeepRecent
```

The osmosis team has changed `baseapp.Set*` => `baseapp.SetSnapshot(store, withInterval(10), withKeepRecent(10)`. I stopped there to let this branch be reviewed first.

### CODE REVIEW

File: x/did/keeper/genesis_test.go [hunk 1]
```diff
@@ -1,16 +1,57 @@
...
+	"github.com/cosmos/cosmos-sdk/std"
 	"github.com/cosmos/cosmos-sdk/types/module"
-	"github.com/tendermint/starport/starport/pkg/cosmoscmd"
+	"github.com/cosmos/cosmos-sdk/x/auth/tx"
 
 	"github.com/elesto-dao/elesto/v2/x/did"
 )
 
+// NOTE: we recreate the encoding config here as we cannot import
+//	 "github.com/elesto-dao/elesto/v2/app" into this test
```
Comment (gsora) [inline x/did/keeper/genesis_test.go:15]: Why is that?

File: x/did/module/module.go [hunk 2]
```diff
@@ -200,15 +200,14 @@ func (am AppModule) RegisterStoreDecoder(sdr sdk.StoreDecoderRegistry) {
...
-	//return simulation.WeightedOperations(
-	//	simState,
-	//	am.keeper,
-	//	am.bk,
-	//	am.ak,
-	//)
-	return nil
+// WeightedOperations returns the all the did module operations with their respective weights.
+func (am AppModule) WeightedOperations(simState module.SimulationState) []simtypes.WeightedOperation {
+	return simulation.WeightedOperations(
```
Comment (gsora) [inline x/did/module/module.go:205]: Why was it commented out before? Does this change makes simulation not work?

**Review by gsora — APPROVED** [review]
I'm approving because this looks 💯 

Left some comments, nothing major!

**Comment by noandrea** [comment]
latest release v0.46.1 of the cosmos-sdk updates the avl dependency to https://github.com/cosmos/iavl/releases/tag/v0.19.1

Does this has any effect in terms of decision making about which sdk to use?

More on a long-term vision, @PaddyMc do you think is a good bet to switch to the osmosis-sdk and depend on it instead of the cosmos-sdk ?

File: x/did/keeper/genesis_test.go (hunk 1 above)
Comment (PaddyMc) [reply x/did/keeper/genesis_test.go:15]: The dreaded import cycle protection in golang, I'm not sure if there's a better way to solve this besides copy/pasta code

File: x/did/module/module.go (hunk 2 above)
Comment (PaddyMc) [reply x/did/module/module.go:205]: I think it was commented out by accident when we were trying to fix simulations :sweat_smile: 

Yeah, the sims should work :+1:

**Comment by PaddyMc** [comment]
> latest release v0.46.1 of the cosmos-sdk updates the avl dependency to https://github.com/cosmos/iavl/releases/tag/v0.19.1
> 
> Does this has any effect in terms of decision making about which sdk to use?
> 
> More on a long-term vision, @PaddyMc do you think is a good bet to switch to the osmosis-sdk and depend on it instead of the cosmos-sdk ?

This PR has no effect on the decision of which SDK to use, and I think that we shouldn't switch at this late stage of the project.

**Review by noandrea — APPROVED** [review]
:100:

**Pull request merged**